File: frontend/types/notification.ts
export interface Notification {
  id: number;
  user_id: number;
  order_id?: number | null;
  title: string;
  message: string;
  type: string;
  is_read: boolean;
  created_at: string;
}

export interface UnreadNotificationCountResponse {
  unread_count: number;
}

export interface MarkAllReadResponse {
  message: string;
  count: number;
}

export interface WebSocketEvent {
  type: string;
  id?: number;
  title?: string;
  message?: string;
  notification_type?: string;
  order_id?: number | null;
  sound?: boolean;
  created_at?: string;
  user_id?: number;
  [key: string]: any;
}

File: backend/app/schemas/notification.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class NotificationResponse(BaseModel):
    id: int
    user_id: int
    order_id: int | None
    title: str
    message: str
    type: str
    is_read: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class UnreadNotificationCountResponse(BaseModel):
    unread_count: int


class MarkAllReadResponse(BaseModel):
    message: str
    count: int
